tests/src/test/scala/org/virtuslab/tests/BazeliskExtension.scala

```diff
@@ -0,0 +1,49 @@
+package org.virtuslab.tests
+
+import com.google.idea.probetests.BazelPluginExtension
+import java.io.{File, FileInputStream}
+import java.net.URL
+import java.nio.file.attribute.PosixFilePermission
+import java.nio.file.{Files, Path}
+import java.util.Collections
+import org.apache.commons.codec.digest.DigestUtils
+import org.apache.commons.io.FileUtils
+import org.apache.commons.lang.SystemUtils
+import org.virtuslab.ideprobe.{Config, IdeProbeFixture, IntelliJFixture}
+
+trait BazeliskExtension extends IdeProbeFixture with BazelPluginExtension{
+  registerFixtureTransformer(_.withAfterWorkspaceSetup((ij: IntelliJFixture, path: Path) => {
+    val c = ij.config
+    installBazelisk(bazelPath(path), ij.config)
+  }))
+
+  def bazelPath(path: Path): Path = {
+    path.resolve("bin/bazel")
+  }
+
+  private def downloadFile(url: URL, filePath: Path, expectedChecksum: String): Unit = {
+    val file: File = filePath.toFile
+    FileUtils.copyURLToFile(
+      url,
+      file
+    )
+    val checksum = DigestUtils.sha256Hex(new FileInputStream(file))
+    if (checksum != expectedChecksum) {
+      throw new RuntimeException(s"Invalid checksum for '$url', expected: $expectedChecksum, actual: $checksum")
+    }
+  }
+
+  private def installBazelisk(bazelExecPath: Path, config: Config): Path = {
+    val (expectedChecksum, url) = if (SystemUtils.IS_OS_MAC_OSX) (
```

Comment:
we have something like `OS.current == OS.Mac` in ide-probe. Also you could just determine config key name from os ("mac" or "linux") to avoid repetition in code